packages/web-frontend/src/components/HierarchyItem.js

```diff
@@ -20,83 +20,73 @@
  * @return {element} a HierarchyItem react component.
  */
 
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import FlatButton from 'material-ui/FlatButton';
 import ClosedIcon from 'material-ui/svg-icons/navigation/chevron-right';
 import OpenIcon from 'material-ui/svg-icons/navigation/expand-more';
-import SelectedIcon from 'material-ui/svg-icons/toggle/radio-button-checked';
-import UnSelectedIcon from 'material-ui/svg-icons/toggle/radio-button-unchecked';
+import SelectedCheckBoxIcon from 'material-ui/svg-icons/toggle/check-box';
+import UnSelectedCheckBoxIcon from 'material-ui/svg-icons/toggle/check-box-outline-blank';
 import CircularProgress from '@material-ui/core/CircularProgress';
 import { ReferenceTooltip } from '@tupaia/ui-components';
 
-export class HierarchyItem extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isOpen: false,
-    };
-  }
-
-  onClick() {
-    if (this.props.onClick) {
-      this.props.onClick();
-    }
-    this.setState(state => ({ isOpen: !state.isOpen }));
-  }
-
-  renderOpenClosedIcon() {
-    const { nestedItems, hasNestedItems } = this.props;
-    const { isOpen } = this.state;
+export const HierarchyItem = React.memo(
```

Comment:
Just double checking that the changes in this file are related to the card's scope? It seems to me that they are not (still worthy refactoring!), but it would be good to confirm for reviewing purposes

Reply:
Yes you are right, it is out of scope of this card, but using class component produces more code comparing to functional component (e.g.: constructor, state and componentDidUpdated). Plus it's also easier if later we want to use useContext() or useReducer().

Reply:
I agree - React functional components are more elegant! Just double checking to make sure I didn't miss reviewing any other aspect of the refactoring 🙂